doc/radosgw/lua-scripting.rst

```diff
@@ -10,6 +10,13 @@ This feature allows users to upload Lua scripts to different context in the rado
 operation was taken, and "postRequest" that will execute after each operation is taken. Script may be uploaded to address requests for users of a specific tenant.
 The script can access fields in the request and modify some fields. All Lua language features can be used in the script.
 
+By default, all lua standard libraries are available in the script, however, in order to allow for other lua modules to be used in the script, we support adding packages to an allowlist:
```

Comment:
How strong do you want this sandbox to be? With the `os` and `debug` packages fully available, any sandbox you create can be trivially broken out of.

Reply:
my intent is not to sandbox the users. since you need radosgw-admin privileges in order to push scripts, i can assume that that the author of the script will not write malicious scripts.
my intent is to allow organizations to control which libraries are safe to use.
I assume that the standard lua libraries are safe, so I allow them by default.

Reply:
> my intent is not to sandbox the users. since you need radosgw-admin privileges in order to push scripts, i can assume that that the author of the script will not write malicious scripts.

The sandbox provided should be documented, at least its philosophy.

> my intent is to allow organizations to control which libraries are safe to use.
> I assume that the standard lua libraries are safe, so I allow them by default.

Lua libraries are not really safe by default. It's expected that an embedding application trims down the libraries to what they need. I would filter out most of the `os` library. all of the `debug` library, `io.popen`, `load`, `loadfile`, `dofile`. Someone very determined could probably break out of the sandbox but there's no reason to make it easy. Keep in mind that someone could be an attacker that gained control of the script somehow.

Also, rather than filtering the libraries, I would create an allowlist of which functions to export and remove the rest.

Reply:
> Lua libraries are not really safe by default. It's expected that an embedding application trims down the libraries to what they need. I would filter out most of the `os` library. all of the `debug` library, `io.popen`, `load`, `loadfile`, `dofile`. Someone very determined could probably break out of the sandbox but there's no reason to make it easy. Keep in mind that someone could be an attacker that gained control of the script somehow.
>

ok. this makes sense, would remove these standard calls:
* `os.exit()`
* `os.execute()`
* `io.popen()`
* `load()`
* `loadfile()`
* `loadstring()`
* `dofile()`
* the debug library

though I'm not 100% sure about `os.execuite()` and `io.popen()` as they may allow for interesting usecases 

> Also, rather than filtering the libraries, I would create an allowlist of which functions to export and remove the rest.

agree that this may be needed, but at this point I would say that whole or nothing approach towards the packages should be sufficient